main_test.go

```diff
@@ -237,50 +363,56 @@ func waitForEtcdReady(clusterName, namespace string, expected int) error {
 	}
 }
 
-func createTestCluster(t *testing.T, clusterName, namespace, version string, numberOfControlPlanes int) {
-	t.Helper()
+func createTestControlPlane(clusterName, namespace, version string, numberOfControlPlanes int) error {
 	if err := setupClusterObject(clusterName, namespace); err != nil {
-		t.Fatal(handleErr(err))
+		return err
 	}
 	for i := 0; i < numberOfControlPlanes; i++ {
-		if err := setupControlPlaneObject(clusterName, namespace, version, i); err != nil {
-			t.Fatal(handleErr(err))
+		if err := createNode(clusterName, controlPlaneNodeType, namespace, version, i); err != nil {
+			return err
 		}
 	}
 	// wait for provider ID on all control planes before creating workerss
 	fmt.Println("Waiting for up to 10 minutes for the provider IDs to appear on the machines")
 	if err := waitForProviderIDOnMachines(clusterName, namespace, numberOfControlPlanes); err != nil {
-		t.Fatal(err)
+		return err
 	}
-	if err := setupWorkerObject(clusterName, namespace, version); err != nil {
-		t.Fatal(handleErr(err))
+	return nil
+}
+
+func createWorkers(t *testing.T, clusterName, namespace, version string, numberOfWorkers int) {
+	t.Helper()
```

Comment:
I find this makes it harder to figure out what broke if it came from in here - wdyt?

Reply:
yeah, i agree with this